feat: Added Layout and basic code to native sign-up

mrudultora opened the pull request:
####  Description
• Created SignupActivity
• Added basic code to _SignupActivity.kt_ and completed layout of _SignupActivity_
• Added related strings in `strings.xml`

#### Related issues
- #3085 

#### Related PRs

#### Screenshots
![img](https://user-images.githubusercontent.com/65870389/104713878-7bbb4900-574a-11eb-9d23-1fd21be0bb8a.jpg)

mrudultora committed:
Added Layout and basic code to native sign-up

teolemon commented:
We should probably have some consent for the TOS and the Open Food Facts newsletter, and possibly support for teams ?

mrudultora commented:
> We should probably have some consent for the TOS and the Open Food Facts newsletter, and possibly support for teams ?

@teolemon Yes sir, that's why I didn't add them in the layout file. Also, would you please tell me the **Api** related info.

mrudultora commented:
Should I add 3 chekbox (each for TOS, newsletter and Professional Account) and one Edittext for Team (optional) ?

teolemon commented:
@mrudultora let's see how it looks like
TOS: https://world.openfoodfacts.org/terms-of-use

teolemon commented:
- [x] I agree to the Open Food Facts terms of use and contribution.
- [x] I am a food producer.
- [x] I'd like to subscribe to the Open Food Facts newsletter (Note: You can unsubscribe from it at any time).

mrudultora committed:
Updated layout and code for Tos and newsLetter

mrudultora commented:
@teolemon  How's this ?

![img](https://user-images.githubusercontent.com/65870389/105504425-c3fbdd80-5ced-11eb-9aff-3d6d494c9144.jpg)

teolemon commented:
@stephanegigandet ? @VaiTon ? What do you think ?
I would rename "Name" into "Display Name", and mention that the username cannot contain spaces, caps or special characters.

mrudultora commented:
@teolemon In which direction shall I proceed now ?

teolemon commented on app/src/main/res/values/strings.xml:
```diff
@@ -1113,6 +1113,19 @@
     <string name="enabled">Enabled</string>
 
+    <!-- Strings related to SignUp -->
+    <string name="signupName">Name</string>
```
```suggestion
    <string name="signupName">Display name</string>
```

teolemon commented on app/src/main/res/values/strings.xml:
```diff
@@ -1113,6 +1113,19 @@
+    <string name="hintConfirmPass">Confirm Password</string>
+    <string name="signup">Sign up</string>
+    <string name="confirm_password_error">Passwords do not match</string>
+    <string name="tos">I agree to the OpenFoodFacts Terms of Use and Contribution.</string>
```
```suggestion
    <string name="tos">I agree to the Open Food Facts terms of use and contribution.</string>
```

teolemon commented on app/src/main/res/values/strings.xml:
```diff
@@ -1113,6 +1113,19 @@
+    <string name="signup">Sign up</string>
+    <string name="confirm_password_error">Passwords do not match</string>
+    <string name="tos">I agree to the OpenFoodFacts Terms of Use and Contribution.</string>
+    <string name="foodProducer">I am a Food Producer.</string>
```
```suggestion
    <string name="foodProducer">I am a food producer.</string>
```

teolemon commented on app/src/main/res/values/strings.xml:
```diff
@@ -1113,6 +1113,19 @@
+    <string name="tos">I agree to the OpenFoodFacts Terms of Use and Contribution.</string>
+    <string name="foodProducer">I am a Food Producer.</string>
+    <string name="subsNewsLetter">I\'d like to subscribe to the Open Food Facts newsletter (Note: You can unsubscribe from it at any time).</string>
+    <string name="acceptTos">Please accept Terms of Use and Contribution</string>
```
```suggestion
    <string name="acceptTos">You need to accept the terms of use and contribution if you want to create an Open Food Facts account</string>
```

mrudultora commented:
@teolemon Is there any minimum length fixed for username ?

mrudultora committed:
Added username condition, regex and changed strings

mrudultora commented:
@teolemon I have made the changes and added a regex check for username. Also mentioned in the layout about the conditions for username. And changed the strings accordingly.

![img](https://user-images.githubusercontent.com/65870389/106359058-0cdb1400-6336-11eb-9df7-dcf3f64207ba.jpg)

teolemon commented:
@stephanegigandet should we ask for the organisation name if the producer box is ticked

mrudultora commented:
@teolemon Any Updates ?

teolemon commented:
Let's add it, since it's in the web form https://world.pro.openfoodfacts.org/cgi/user.pl
![image](https://user-images.githubusercontent.com/1689815/107032525-44254700-67b4-11eb-8263-9d0b8c57eae2.png)

mrudultora committed:
Brand Name if producer box is ticked

mrudultora commented:
@teolemon Added. Producer or Brand name textInputEditText will be visible when food producer checkbox is ticked and hides when that checkbox is unticked.
Also, will you please provide me the drawables for username, food producer, email. I will replace them in textInputEditText.

![img](https://user-images.githubusercontent.com/65870389/107329237-e192ba80-6ad5-11eb-95fa-3162a567e077.jpg)

mrudultora commented:
@teolemon If you don't mind, would you please guide me on completing this sign-up process with API related part. Till now in this draft PR, I worked on UI and basic code. But, I want to complete the kotlin code also ;-)

teolemon commented on app/src/main/AndroidManifest.xml:
```diff
@@ -2,26 +2,20 @@
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
-
-    <!-- Merging strategy for the imported manifests -->
-    <uses-sdk tools:overrideLibrary="io.sentry.android" />
```
Should sentry disappear ?

mrudultora replied:
@teolemon No sentry didn't disappeard. I formatted the code of **AndroidManifest**. So it's just some lines changed there position. If possible please check the line no. 6

mrudultora replied:
Please have a look at this screenshot.

![Untitled](https://user-images.githubusercontent.com/65870389/107643295-f6b94600-6c9b-11eb-8829-ea2144f341c4.png)

mrudultora commented:
Now, I guess just small part is remaining. So, please provide me the url of API for signup so that this native signup is implemented soon. @teolemon @VaiTon

teolemon commented:
@mrudultora there's no API. You need to create and send the form, like the webform does: https://world.pro.openfoodfacts.org/cgi/user.pl
```
<form method="post" action="/cgi/user.pl" enctype="multipart/form-data">

Name <input type="text" name="name" size="80" autocomplete="name" id="name">
E-mail address</td><td><input type="text" name="email" size="80" autocomplete="email">
User name<br><span class="info">(non-accented letters, digits and/or dashes)</span></td><td>
<input type="text" name="userid" size="40" autocomplete="username" id="userid" onkeyup="normalize_string_value(this)">

Password</td><td><input type="password" name="password" autocomplete="new-password">
Confirm password<input type="password" name="confirm_password" autocomplete="new-password"></td></tr>

Professional account
If you work for a producer or brand and will add or complete data for your own products only, you can get access to our completely free Platform for Producers.
The platform for producers allows manufacturers to easily import data and photos for all their products, to mark them as official, and to get free analysis of improvement opportunities for their products.

<input type="checkbox" id="pro" name="pro" checked="">
<input type="hidden" name="pro_checkbox" value="1"> This is a producer or brand account.

Producer or brand: <input type="text" name="requested_org" id="requested_org">
Please enter the name of your organization (company name or brand).

<input type="checkbox" name="newsletter" value="on" checked="checked">Subscribe to the newsletter (2 emails per month maximum)
You can unsubscribe from the lists at any time.

<input type="hidden" name="action" value="process">
<input type="hidden" name="type" value="add">         
  
<input type="submit" name=".submit" class="button">

</form>
```

VaiTon commented:
@teolemon can we get an endpoint to perform the POST request on?

teolemon commented:
@VaiTon @mrudultora I've raised the topic in #productopener, but we could so without for this version (by just posting an html form from the app).

teolemon committed:
Merge branch 'develop' into NativeSignupDraftPR

teolemon commented:
HTTP POST
Endpoint: `https://world.openfoodfacts.org/cgi/user.pl`
Encode: `multipart/form-data`
Fields:
* `name` - name
* `email` - email address 
* `userid` - user name (non-accented letters, digits and/or dashes) (Add a normalizing function):
* `password`
* `confirm_password` - confirm password

Professional account
If you work for a producer or brand and will add or complete data for your own products only, you can get access to our completely free Platform for Producers.
The platform for producers allows manufacturers to easily import data and photos for all their products, to mark them as official, and to get free analysis of improvement opportunities for their products.

```
<input type="checkbox" id="pro" name="pro" checked="">
<input type="hidden" name="pro_checkbox" value="1"> This is a producer or brand account.
```

Additional fields:
* `requested_org` - producer or brand
Please enter the name of your organization (company name or brand).
* `newsletter=on` or nothing
Subscribe to the newsletter (2 emails per month maximum)
You can unsubscribe from the lists at any time.
* `action=process` (Hidden)
* `type=add` (Hidden)      
```
<input type="submit" name=".submit" class="button">
```

VaiTon commented:
> `action=process` (Hidden)
`type=add` (Hidden)

Are these mandatory for every account or only for producer accounts?

teolemon commented:
@VaiTon I believe that kind of thing is for all accounts. It's something @stephanegigandet adds frequently to his forms

teolemon committed:
Merge branch 'develop' into pr/3737

teolemon committed:
Merge branch 'develop' into pr/3737

teolemon commented:
@mrudultora fixed the merge conflicts and rebased

teolemon committed:
Merge branch 'develop' into NativeSignupDraftPR

teolemon commented:
@mrudultora @VaiTon fixed the merge conflicts and rebased

teolemon committed:
Merge branch 'develop' into NativeSignupDraftPR

teolemon commented:
rebased @mrudultora @VaiTon Can we go the simple route, and use the current url to send the form ?
`https://world.openfoodfacts.org/cgi/user.pl`

teolemon committed:
Merge branch 'develop' into pr/3737

teolemon committed:
Merge branch 'NativeSignupDraftPR' of https://github.com/mrudultora/openfoodfacts-androidapp into pr/3737

teolemon committed:
Merge branch 'develop' into pr/3737

teolemon committed:
Merge branch 'develop' into pr/3737

teolemon commented:
i've rebased on latest develop, but this PR currently doesn't use databinding. As a result: 
```
Execution failed for task ':app:mergeOffPlaystoreScreenshotsResources'.
> /home/runner/work/openfoodfacts-androidapp/openfoodfacts-androidapp/app/src/main/res/layout/activity_signup.xml: Error: ERROR: Found <layout> but data binding is not enabled.
  
  Add buildFeatures.dataBinding = true to your build.gradle to enable it. file:///home/runner/work/openfoodfacts-androidapp/openfoodfacts-androidapp/app/src/main/res/layout/activity_signup.xml
```

teolemon committed:
Merge branch 'develop' into NativeSignupDraftPR

teolemon committed:
Merge branch 'develop' into NativeSignupDraftPR
